centrapay/centrapay-docs
# Remove vue-cli dependencies

This started with an attempt to upgrade Astro to v3.0
- Astro v3.0 no longer supports Node 16. We need to upgrade our tooling to Node 18. See https://docs.astro.build/en/guides/upgrade-to/v3/#removed-support-for-node-16
- vue-cli contains a dependency that is not compatible with Node 18 (@achrinza/node-ipc@9). See https://github.com/vuejs/vue-cli/issues/7116

This compatible issue has been fixed but Centrapay Docs does not need vue-cli and the module itself is in maintenance mode. See https://cli.vuejs.org/

Removing these vue-cli dependencies required a change to use eslint directly for linting instead. We are now using eslint-plugin-vue (it was already installed) as per Vue's official guidelines. See https://vuejs.org/guide/scaling-up/tooling.html#linting

I've also added init: true to the docker compose config to forward signals correctly.

See our dev environment for a live preview - http://centrapay-docs.dev.s3-website-ap-southeast-1.amazonaws.com

Test plan:
- Confirm Eslint is linting Vue files using its --debug flag
- Confirm that the local dev server runs correctly
- Confirm that the static Centrapay Docs site builds correctly
- Confirm the site navigation still works
